### PR TITLE
avoid cmake syntax error due to "\" path separators on windows

### PR DESCRIPTION
**Pull request by NNemec**

### CODE REVIEW

File: mesonbuild/cmake/executor.py [hunk 1]
```diff
@@ -236,7 +236,7 @@ def choose_compiler(lang: str):
 set(CMAKE_CXX_IGNORE_EXTENSIONS inl;h;hpp;HPP;H;o;O;obj;OBJ;def;DEF;rc;RC)
 set(CMAKE_CXX_SOURCE_FILE_EXTENSIONS C;M;c++;cc;cpp;cxx;mm;CPP)
 set(CMAKE_SIZEOF_VOID_P "{}")
-'''.format(cxx_comp, cxx_launcher, ctypes.sizeof(ctypes.c_voidp)))
+'''.format(cxx_comp.replace('\\','/'), cxx_launcher.replace('\\','/'), ctypes.sizeof(ctypes.c_voidp)))
```
Comment (ignatenkobrain) [inline mesonbuild/cmake/executor.py]: __[Flake8]__ [E231] missing whitespace after ',' ([link](https://sider.review/gh/repos/19784232/pulls/5763#issue-4408906))

<sub>posted by Sider</sub>

File: mesonbuild/cmake/executor.py [hunk 2]
```diff
@@ -221,7 +221,7 @@ def choose_compiler(lang: str):
 set(CMAKE_C_SOURCE_FILE_EXTENSIONS c;m)
 set(CMAKE_C_IGNORE_EXTENSIONS h;H;o;O;obj;OBJ;def;DEF;rc;RC)
 set(CMAKE_SIZEOF_VOID_P "{}")
-'''.format(c_comp, c_launcher, ctypes.sizeof(ctypes.c_voidp)))
+'''.format(c_comp.replace('\\','/'), c_launcher.replace('\\','/'), ctypes.sizeof(ctypes.c_voidp)))
```
Comment (ignatenkobrain) [inline mesonbuild/cmake/executor.py]: __[Flake8]__ [E231] missing whitespace after ',' ([link](https://sider.review/gh/repos/19784232/pulls/5763#issue-4408905))

<sub>posted by Sider</sub>

**Comment by scivision** [comment]
When does this issue arise? When using CMake subproject? Reason being, I would like to see a simple broken/fixed test case, as this is probably more generally addressed by using pathlib.Path.

**Comment by mensinda** [comment]
@scivision This fix seems to be for the CMake dependency backend. 

@NNemec For the test case, could you try if the error also occurs with the test case `test cases/linuxlike/13 cmake dependency` on Windows? If it does, then moving this testcase to `test cases/cmake` might be enough for the test case (this way the test is also executed on the Windows CI).

**Comment by NNemec** [comment]
@scivision The issue arises in a rather complex situation: building two interdependent meson-based libraries inside the vcpkg system within a VS2019 Developer PowerShell. The changes in vcpkg necessary to get to this point at all are work in progress. Stripping down this scenario to something I could put into a test case will take time that I cannot invest at this moment, especially since I have very limited understanding of all the things meson is doing.

@mensinda Thanks for your pointer. Unfortunately, '13 cnake dependency' tries to find ZLIB. There is a comment '# Zlib is probably on all dev machines.' I have no idea how to install zlib on a Windows machine in such a way that it can be found by mesonbuild.

So overall: the problem and the fix are quite obvious. How to create a test case is not obvious at all. :-(

**Comment by mensinda** [comment]
Even without a test case (if you cannot reproduce the bug without vcpkg) this should be merged. Maybe you could use a more platform-independent [approach](https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.as_posix) and fix the flake8 errors.

**Comment by NNemec** [comment]
@mensinda thanks for the hint. I cleaned up the patch, added a comment and tested it carefully in my local setup.

**Comment by mensinda** [comment]
CI failure seems important

**Comment by NNemec** [comment]
It does indeed. For some reason, pathlib.PurePath does not appear to do the right thing on cygwin.  My guess is that the environment variables contain paths in Windows format, but cygwin tries to interpret them as Posix path. If that is true, my original approach of manual normalization without making assumptions would have been more appropriate. (in fact, that old approach did work for all CI tests.)

**Pull request merged**